File: 30DaysOfCode/Day6/LetsReview.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    int t;
    cin >> t;
    while (t) {
        string input;
        cin >> input;
        
        vector<char> splitInput( input.begin(), input.end() );
        
        //Printing out the even indexed characters
        for (int i = 0; i < splitInput.size(); i = i + 2) {
            cout << splitInput[i];
        }
        
        cout << " ";
        
        //Printing out the odd indexed characters
        for (int i = 1; i < splitInput.size(); i = i + 2) {
            cout << splitInput[i];
        }
        
        cout << endl;
        t--;
    }
    return 0;
}

File: 30DaysOfCode/Day6/LetsReview.java
import java.io.*;
import java.util.*;

public class LetsReview {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        
        Scanner scan = new Scanner(System.in);
        
        int t;
        String input;
        
        t = scan.nextInt();
        input = scan.nextLine();
        
        while (t>0) {
            
            input = scan.nextLine();
            
            char[] splitInput = input.toCharArray();
            
            //Printing out the even indexed characters
            for (int i=0; i<input.length(); i=i+2) {
                System.out.print(splitInput[i]);
            }
            
            System.out.print(" ");
            
            //Printing out the odd indexed characters
            for (int i=1; i<input.length(); i=i+2) {
                System.out.print(splitInput[i]);
            }
            
            System.out.println("");
            
            t--;
        }
        
        scan.close();
    }
}